core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -332,7 +333,7 @@ public int numConnections() {
      * Waits until the result of {@link CompletableFuture} which is completed after the {@link #close()} or
      * {@link #closeAsync()} operation is completed.
      */
-    public void blockUntilShutdown() throws Exception {
+    public void blockUntilShutdown() throws ExecutionException, InterruptedException {
         whenClosed().get();
     }
```

Comment:
After thinking a little bit, how about catching `ExecutionException` and turning it into `CompletionException`? i.e.

```suggestion
    public void blockUntilShutdown() throws InterruptedException {
        try {
            whenClosed().get();
        } catch (ExecutionException e) {
            throw new CompletionException(e.toString(), Exceptions.peel(e));
        }
    }
```

.. because we do not usually use `ExecutionException` in our API.

Reply:
Yes I can change that. It makes sense to me.